#include "mainwindow.h"
#include "ui_mainwindow.h"
#include "QMessageBox"

MainWindow::MainWindow(QWidget *parent) :
    QMainWindow(parent),
    ui(new Ui::MainWindow)
{
    ui->setupUi(this);

    connect(ui->Inf_b,SIGNAL(clicked(bool)),this,SLOT(func1()));
    connect(ui->exit_b,SIGNAL(clicked(bool)),this,SLOT(func2()));
    connect(ui->push1,SIGNAL(clicked(bool)),this,SLOT(func3()));
    connect(ui->push2,SIGNAL(clicked(bool)),this,SLOT(func4()));
    connect(ui->push3,SIGNAL(clicked(bool)),this,SLOT(func5()));
}

MainWindow::~MainWindow()
{
    delete ui;
}

void MainWindow::func5()
{
    QString new_lable5;


    double vess = (ui->doubleSpinBox_6->value());

    double Timee  = (ui->doubleSpinBox_7->value());

    double Speed  = (ui->doubleSpinBox_8->value());

    double res0 = ((((18*Speed-20)*vess)/1000)/60)*Timee;

    new_lable5 = QString::number(res0);
    if (Speed <= 1,2)
    {
        QMessageBox::warning(this,"Предупреждение!!","Данное значение скорости не поддерживается формулой, введите большее значение");
    }
    else
    {
    if(res0 == 0)
    {
        QMessageBox::warning(this,"Предупреждение!!","Неправильные данные, одно или несколько полей пусты!");

    }
    else if ((res0>0) && (res0<=100))
    {
        QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable5 +" ккал"+ "\nХорошая пробежка, но вам следует постепенно увеличивать нагрузку! ");

    }
    else if ((res0>100) && (res0<=200))
    {
        QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable5 +" ккал"+ "\nЭто была хорошая работа, продолжайте совершенствовать себя! ");

    }
    else if ((res0>200) && (res0<=400))
    {
        QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable5 +" ккал"+ "\nВы хорошо потренировались, не останавливайтесь на достигнутом! ");

    }
    else if ((res0>400) && (res0<=800))
    {
        QMessageBox::about(this,"Результат расчета","                       Ваш результат = "+new_lable5 +" ккал"+ "\nОтличный результат! Продолжайте в том же духе! ");

    }
    else if ((res0>800))
    {
        QMessageBox::about(this,"Результат рассчета","                              Ваш результат = "+new_lable5 +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере? ");

    }
}
}

void MainWindow::func1()
{

    QMessageBox::information(this,"Информация о программе","     Данная программа предназначена для рассчитывания количества калорий, которые человек сжигает при беге.\n      В зависимости от количества данных, которыми обладает пользователь, представлены три формулы.\n      Первая, пожалуй, является наименее точной, однако для нее требуется самое минимальное количество данных, третья и вторая требуют для расчета больше информации, но их точность значительно выше.\n      Данная программа реализует каждую из формул по желанию.");

}

void MainWindow::func2()
{
   QApplication::quit();

}



void MainWindow::func3()
{
    QString new_lable;

    double ves1 = (ui->doubleSpinBox->value());

    double Ras1 = (ui->doubleSpinBox_2->value());

    double result = ves1*Ras1;
    new_lable = QString::number(result);
   // QMessageBox::about(this,"Результат рассчета",new_lable);

    if(result == 0)
    {
        QMessageBox::warning(this,"Предупреждение!!","Неправильные данные, одно или несколько полей пусты!");

    }
    else if ((result>0) && (result<=100))
    {
        QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable +" ккал"+ "\nХорошая пробежка, но вам следует постепенно увеличивать нагрузку! ");

    }
    else if ((result>100) && (result<=200))
    {
        QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable +" ккал"+ "\nЭто была хорошая работа, продолжайте совершенствовать себя! ");

    }
    else if ((result>200) && (result<=400))
    {
        QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable +" ккал"+ "\nВы хорошо потренировались, не останавливайтесь на достигнутом! ");

    }
    else if ((result>400) && (result<=800))
    {
        QMessageBox::about(this,"Результат расчета","                       Ваш результат = "+new_lable +" ккал"+ "\nОтличный результат! Продолжайте в том же духе! ");

    }
    else if ((result>800))
    {
        QMessageBox::about(this,"Результат рассчета","                              Ваш результат = "+new_lable +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере? ");

    }
}

void MainWindow::func4()
{
    QString new_lable1;
    QString new_lable2;

    double ves2 = (ui->doubleSpinBox_3->value());

    double Pulse  = (ui->doubleSpinBox_4->value());

    double Time  = (ui->doubleSpinBox_5->value());

    double res2 = (0.014*ves2*Time*(0.12*Pulse-7))/60;

    new_lable1 = QString::number(res2);
    new_lable2 = QString::number(Pulse);
    if (Pulse<=58,4)
    {
            QMessageBox::warning(this,"Предупреждение!!","Это значение пульса не поддерживается формулой, введите большее значениe!");

    }
    else
    {
    if(res2 == 0)
    {
        QMessageBox::warning(this,"Предупреждение!!","Неправильные данные, одно или несколько полей пусты!");

    }
    else if ((res2>0) && (res2<=100))
    {

        if ((Pulse>0) && (Pulse<=140))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nХорошая пробежка, но вам следует постепенно увеличивать нагрузку!"+"\nВаш пульс = "+new_lable2+". Это хорошее значение пульса!");
        }
        if ((Pulse>140) && (Pulse<=160))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nХорошая пробежка, но вам следует постепенно увеличивать нагрузку!"+"\nВаш пульс = "+new_lable2+". Ваш пульс повышен, но допустим!");
        }
        if ((Pulse>160) && (Pulse<=180))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nХорошая пробежка, но вам следует постепенно увеличивать нагрузку!"+"\nВаш пульс = "+new_lable2+". Довольно высокий пульс, рекомендуется снизить темп бега!");
        }
        if (Pulse>180)
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nХорошая пробежка, но вам следует постепенно увеличивать нагрузку!"+"\nВаш пульс = "+new_lable2+". Чрезмерно высокй пульс, немодленно снизьте нагрузку, иначе есть риск травмы!");
        }
    }


    else if ((res2>100) && (res2<=200))
    {

        if ((Pulse>0) && (Pulse<=140))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nЭто была хорошая работа, продолжайте совершенствовать себя!"+"\nВаш пульс = "+new_lable2+". Это хорошее значение пульса!");
        }
        if ((Pulse>140) && (Pulse<=160))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nЭто была хорошая работа, продолжайте совершенствовать себя!"+"\nВаш пульс = "+new_lable2+". Ваш пульс повышен, но допустим!");
        }
        if ((Pulse>160) && (Pulse<=180))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nЭто была хорошая работа, продолжайте совершенствовать себя!"+"\nВаш пульс = "+new_lable2+". Однако, пульс достаточно высок, рекомендуется немного снизить темп бега!");
        }
        if (Pulse>180)
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nЭто была хорошая работа, продолжайте совершенствовать себя!"+"\nВаш пульс = "+new_lable2+". Чрезмерно высокй пульс, немедленно снизьте нагрузку, иначе есть риск травмы!");
        }
    }




    else if ((res2>200) && (res2<=400))
    {

        if ((Pulse>0) && (Pulse<=140))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы хорошо потренировались, не останавливайтесь на достигнутом!"+"\nВаш пульс = "+new_lable2+". Это хорошее значение пульса!");
        }
        if ((Pulse>140) && (Pulse<=160))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы хорошо потренировались, не останавливайтесь на достигнутом!"+"\nВаш пульс = "+new_lable2+". Ваш пульс повышен, но допустим!");
        }
        if ((Pulse>160) && (Pulse<=180))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы хорошо потренировались, не останавливайтесь на достигнутом!"+"\nВаш пульс = "+new_lable2+". Однако, пульс достаточно высок, рекомендуется немного снизить темп бега!");
        }
        if (Pulse>180)
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы хорошо потренировались, не останавливайтесь на достигнутом!"+"\nВаш пульс = "+new_lable2+". Чрезмерно высокй пульс, немедленно снизьте темп, иначе есть риск травмы!");
        }
    }


    else if ((res2>400) && (res2<=800))
    {

        if ((Pulse>0) && (Pulse<=140))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nОтличный результат! Продолжайте в том же духе!"+"\nВаш пульс = "+new_lable2+". Это хорошее значение пульса!");
        }
        if ((Pulse>140) && (Pulse<=160))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nОтличный результат! Продолжайте в том же духе!"+"\nВаш пульс = "+new_lable2+". Ваш пульс повышен, но допустим!");
        }
        if ((Pulse>160) && (Pulse<=180))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nОтличный результат! Продолжайте в том же духе!"+"\nВаш пульс = "+new_lable2+". Однако, пульс достаточно высок, рекомендуется немного снизить темп бега!");
        }
        if (Pulse>180)
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nОтличный результат! Продолжайте в том же духе!"+"\nВаш пульс = "+new_lable2+". Чрезмерно высокй пульс, немедленно снизьте темп, иначе есть риск травмы!");
        }
    }


    else if ((res2>800))
    {
        //QMessageBox::about(this,"Результат рассчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере? ");
        if ((Pulse>0) && (Pulse<=140))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере?"+"\nВаш пульс = "+new_lable2+". Это хорошее значение пульса!");
        }
        if ((Pulse>140) && (Pulse<=160))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере?"+"\nВаш пульс = "+new_lable2+". Ваш пульс повышен, но допустим!");
        }
        if ((Pulse>160) && (Pulse<=180))
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере?"+"\nВаш пульс = "+new_lable2+". Однако, пульс достаточно высок, рекомендуется немного снизить темп бега!");
        }
        if (Pulse>180)
        {
          QMessageBox::about(this,"Результат расчета","                              Ваш результат = "+new_lable1 +" ккал"+ "\nВы просто невероятны! Мoжет следует задуматься о префессиональной карьере?"+"\nВаш пульс = "+new_lable2+". Чрезмерно высокй пульс, немедленно снизьте темп, иначе есть риск травмы!");
        }
    }

}


}
